### ISSUE TITLE
RecordInput blocks if file_pattern returns no files.

### ISSUE THREAD

**Issue by TimZaman** [tensorflow/tensorflow#11372]
The `tensorflow.python.ops.data_flow_ops.RecordInput()` blocks forever if the first input argument with keyword `file_pattern` is a pattern that does not point to (an) existing file(s).

It does not seem to be waiting for that file to appear either, because if I make it appear after the launch it's still blocked, leading me to believe this is a bug.

Reproduce:
```python
sess = tf.Session()
record_input = data_flow_ops.RecordInput(file_pattern="foo")
yield_op = record_input.get_yield_op()
sess.run(tf.global_variables_initializer())
sess.run(yield_op)
```

**Comment by michaelisard** [comment]
@ekelsen it looks as if RecordYielder main loop will exit immediately with an error if the fileset is empty, which should cause the yield op to return immediately with an error. Not sure why the above behavior is being reported.

**Comment by TimZaman** [comment]
Version is 1.2.1 from pypi (Both macOS and Ubuntu14.04).

**Comment by ekelsen** [comment]
Bug identified, should be fixed internally soon.

**Comment by TimZaman** [comment]
Thank you sir. Can you check out if you can fix #11396 while you're at it?
